lib/mongoid/attributes/nested.rb

```diff
@@ -76,7 +79,8 @@ def accepts_nested_attributes_for(*args)
         #
         # @since 3.1.4
         def autosave_nested_attributes(association)
-          if association.autosave?
+          if association.autosave? || (association.options[:autosave].nil? && !association.embedded?)
+            association.options[:autosave] = true
```

Comment:
What is the purpose of this assignment?

Reply:
In order for the autosave functionality to work properly, the association needs to be marked as autosave despite the fact that the option isn't present. Because the method `Association#autosave?` (which is used elsewhere in the codebase) is implemented by checking the autosave option, this is the most straightforward way to mark it.

Reply:
Thanks, can you put this entire comment in the source then?